game/recastnavigation_bin.ksy

```diff
@@ -0,0 +1,283 @@
+meta:
+  id: recastnavigation_bin
+  file-extension: bin
+  license: CC0-1.0
+  endian: le
+doc: this spec can be used to parse recastnavigation binary files
+doc-ref: https://masagroup.github.io/recastdetour
+seq:
+  - id: header
+    type: navmesh_set_header
+  - id: tiles
+    type: navmesh_tile
+    repeat: expr
+    repeat-expr: header.num_tiles
+types:
+  navmesh_set_header:
+    seq:
+      - id: magic
+        contents: "TESM"
+      - id: version
+        type: s4
+      - id: num_tiles
+        type: s4
+      - id: params
+        type: dt_navmesh_params
+  dt_navmesh_params:
+    seq:
+      - id: orig
+        type: float3
+        doc: The world space origin of the navigation mesh's tile space. [(x, y, z)]
+      - id: tile_width
+        type: f4
+        doc: The width of each tile. (Along the x-axis.)
+      - id: tile_height
+        type: f4
+        doc: The height of each tile. (Along the z-axis.)
+      - id: max_tiles
+        type: s4
+        doc: The maximum number of tiles the navigation mesh can contain.
+      - id: max_polys
+        type: s4
+        doc: The maximum number of polygons each tile can contain.
+    doc: |
+      Configuration parameters used to define multi-tile navigation meshes.
+      The values are used to allocate space during the initialization of a navigation mesh.
+  float3:
+    seq:
+      - id: x
+        type: f4
+      - id: y
+        type: f4
+      - id: z
+        type: f4
+  dt_tile_ref:
+    seq:
+      - id: ref
+        type: u4
+  navmesh_tile_header:
```

Comment:
Same thing about this type: you don't really need this header separately, it can be safely embedded into `navmesh_tile`.